service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AgentPersonMapper.java

```diff
@@ -47,16 +47,25 @@ public String mapAgentPerson(AgentDirectory.AgentKey agentKey, String agentDirec
             if (practitionerGiven.isEmpty() && practitionerFamily.isEmpty()) {
                 builder.practitionerFamilyName(UNKNOWN);
             }
+
+            messageContext.getInputBundleHolder()
+                .getResource(new IdType(agentKey.getOrganizationReference()))
+                .map(Organization.class::cast)
+                .map(OrganizationToAgentMapper::mapOrganizationToAgentInner)
+                .ifPresent(builder::organization);
+        } else {
```

Comment:
I see that this is using `practitioner.isPresent()`. If working with `Optional` it's usually preferable to use `.map(...).orElse(...)`. If it becomes too big, then extract bits into static functions

Reply:
Example:
```
            .ifPresentOrElse(
                practitioner -> {
                    builder.practitioner(true);
                    buildPractitionerPrefix(practitioner).ifPresent(builder::practitionerPrefix);

                    var practitionerGiven = buildPractitionerGivenName(practitioner);
                    var practitionerFamily = buildPractitionerFamilyName(practitioner);

                    practitionerGiven.ifPresent(builder::practitionerGivenName);
                    practitionerFamily.ifPresent(builder::practitionerFamilyName);

                    if (practitionerGiven.isEmpty() && practitionerFamily.isEmpty()) {
                        builder.practitionerFamilyName(UNKNOWN);
                    }

                    messageContext.getInputBundleHolder()
                        .getResource(new IdType(agentKey.getOrganizationReference()))
                        .map(Organization.class::cast)
                        .map(OrganizationToAgentMapper::mapOrganizationToAgentInner)
                        .ifPresent(builder::organization);
                },
                () -> messageContext.getInputBundleHolder()
                    .getResource(new IdType(agentKey.getOrganizationReference()))
                    .map(Organization.class::cast)
                    .map(Organization::getName)
                    .ifPresent(organizationName -> {
                        builder.practitioner(true);
                        builder.practitionerFamilyName(organizationName);
                    }));
```